src/frequenz/client/dispatch/_cli_types.py

```diff
@@ -32,12 +32,15 @@ def __init__(self) -> None:
 
     def convert(
         self, value: Any, param: click.Parameter | None, ctx: click.Context | None
-    ) -> datetime:
-        """Convert the value to a datetime object."""
-        if isinstance(value, datetime):
-            return value
+    ) -> datetime | Literal["NOW"] | None:
+        """Convert the value to a datetime object or the string "NOW"."""
+        if isinstance(value, datetime) or value == "NOW":
+            return cast(datetime | Literal["NOW"], value)
```

Comment:
Since you are also returning `Literal["NOW"]` below to do a more relaxed comparison to `"NOW"`, wouldn't it be easier and avoid the cast to just do this here?
```suggestion
        if isinstance(value, datetime):
            return value
```